app/src/lib/editors/darwin.ts

```diff
@@ -182,150 +143,27 @@ async function findApplication(editor: ExternalEditor): Promise<string | null> {
-    findApplication(ExternalEditor.TextMate),
-    findApplication(ExternalEditor.Brackets),
-    findApplication(ExternalEditor.WebStorm),
-    findApplication(ExternalEditor.Typora),
-    findApplication(ExternalEditor.CodeRunner),
-    findApplication(ExternalEditor.SlickEdit),
-    findApplication(ExternalEditor.IntelliJ),
-    findApplication(ExternalEditor.Xcode),
-    findApplication(ExternalEditor.GoLand),
-    findApplication(ExternalEditor.AndroidStudio),
-    findApplication(ExternalEditor.Rider),
-    findApplication(ExternalEditor.Nova),
-  ])
-
-  if (atomPath) {
-    results.push({ editor: ExternalEditor.Atom, path: atomPath })
-  }
-
-  if (macVimPath) {
-    results.push({ editor: ExternalEditor.MacVim, path: macVimPath })
-  }
-
-  if (codePath) {
-    results.push({ editor: ExternalEditor.VSCode, path: codePath })
-  }
-
-  if (codeInsidersPath) {
-    results.push({
-      editor: ExternalEditor.VSCodeInsiders,
-      path: codeInsidersPath,
-    })
-  }
-
-  if (codiumPath) {
-    results.push({ editor: ExternalEditor.VSCodium, path: codiumPath })
-  }
-
-  if (sublimePath) {
-    results.push({ editor: ExternalEditor.SublimeText, path: sublimePath })
-  }
-
-  if (bbeditPath) {
-    results.push({ editor: ExternalEditor.BBEdit, path: bbeditPath })
-  }
-
-  if (phpStormPath) {
-    results.push({ editor: ExternalEditor.PhpStorm, path: phpStormPath })
-  }
-
-  if (pyCharmPath) {
-    results.push({ editor: ExternalEditor.PyCharm, path: pyCharmPath })
-  }
-
-  if (rubyMinePath) {
-    results.push({ editor: ExternalEditor.RubyMine, path: rubyMinePath })
-  }
-
-  if (textMatePath) {
-    results.push({ editor: ExternalEditor.TextMate, path: textMatePath })
-  }
-
-  if (bracketsPath) {
-    results.push({ editor: ExternalEditor.Brackets, path: bracketsPath })
-  }
-
-  if (webStormPath) {
-    results.push({ editor: ExternalEditor.WebStorm, path: webStormPath })
-  }
-
-  if (typoraPath) {
-    results.push({ editor: ExternalEditor.Typora, path: typoraPath })
-  }
-
-  if (codeRunnerPath) {
-    results.push({ editor: ExternalEditor.CodeRunner, path: codeRunnerPath })
-  }
-
-  if (slickeditPath) {
-    results.push({ editor: ExternalEditor.SlickEdit, path: slickeditPath })
-  }
-
-  if (intellijPath) {
-    results.push({ editor: ExternalEditor.IntelliJ, path: intellijPath })
-  }
-
-  if (xcodePath) {
-    results.push({ editor: ExternalEditor.Xcode, path: xcodePath })
-  }
-
-  if (golandPath) {
-    results.push({ editor: ExternalEditor.GoLand, path: golandPath })
-  }
-
-  if (androidStudioPath) {
-    results.push({
-      editor: ExternalEditor.AndroidStudio,
-      path: androidStudioPath,
-    })
-  }
-
-  if (riderPath) {
-    results.push({ editor: ExternalEditor.Rider, path: riderPath })
-  }
-
-  if (novaPath) {
-    results.push({ editor: ExternalEditor.Nova, path: novaPath })
+  const results: Array<IFoundEditor<string>> = []
+
+  const editorPaths = await Promise.all(
```

Comment:
Similar comment here as with the win32 `getAvailableEditors`. I think we should move the call to `findApplication` into the loop below.